### PR TITLE
Add support for swift modules with @objc interfaces

### PR DESCRIPTION
**Pull request by vasvf**
This adds .swiftinterface files to artifacts that are needed for swift-only modules that contain @objc exposing interfaces.

Also tried to modify tests to fit this new type of file, but maybe I missed something

### CODE REVIEW

**Review by polac24 — COMMENTED** [review]
That seems to be a reasonable change that is useful for projects with enabled `BUILD_LIBRARY_FOR_DISTRIBUTION`.  In the PR description you mentioned adding `@objc public ...` but as far as I am aware, generating .swiftinterface depends for swift-evolution enabled builds (In Xcode managed by `BUILD_LIBRARY_FOR_DISTRIBUTION`).

Owing to that, you PR implementation seems to be fine but we should have two unit tests (in change `BuildArtifactCreatorTests.swift` and `ArtifactSwiftProductsBuilderImplTests.swift`:
* Mode without swift-evolution - that is the current stat of these tests
* mode with swift-evolution:  same as above + expect zipping and unzipping extra `.swiftinterface` 

In other words, your unit testing changes should be applied in a separate functions.

**Comment by vasvf** [comment]
Separated tests in two files you specified.

I also removed unused `let swiftmoduleExtensionsToInclude` in `ArtifactSwiftProductsBuilder.swift`

I also added swiftinterface file check in swiftcTests

**Review by polac24 — APPROVED** [review]
Thanks!
